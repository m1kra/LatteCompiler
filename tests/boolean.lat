
void main(){
    boolean a = true, b = false;
    printBool(a && b); printBool(a || b);
    if ((((a || b) && (1 != 0))) && !b){
        printString("Ok.");
    }
    printString("&&");
    printBool(-1 > 0); printBool(123 < 321);
    printBool(-3 == -3); printBool(-3 != -3);
    printBool(1 <= -1); printBool(-1 >= 1);
}

void printBool(boolean b){
    if (b)
        printInt(1);
    else
        printInt(0);
}

