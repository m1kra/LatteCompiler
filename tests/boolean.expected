0
1
Ok.
&&
0
1
1
0
0
0